### ISSUE TITLE
Avoid automated behaviours in the Navigation block when in preview mode

### ISSUE THREAD

**Issue by scruffian** [WordPress/gutenberg#45887]
In https://github.com/WordPress/gutenberg/pull/44173 we introduced a fallback to classic menus. This is also triggered when the block is previewed in a pattern browser. We should detect whether we are in a preview state, using an `__unstableIsPreviewMode` attribute ( introduced [here](https://github.com/WordPress/gutenberg/pull/44784)), and if we are then we shouldn't import the classic menu.

**Comment by getdave** [comment]
This feels immediately actionable.

Next steps:

- verify issue still occurs
- implement conditionals to avoid running automated behaviours when in preview mode

cc'ing @ntsekouras who originally implemented `__unstableIsPreviewMode` and may have information on its stability and utility in this scenario.

**Comment by ntsekouras** [comment]
Yeah, the flag was actually stabilized recently [here](https://github.com/WordPress/gutenberg/pull/66149) and the name is `isPreviewMode` now.

**Comment by getdave** [comment]
Having looking into this I'm not entirely clear on the bug we're solving here. We can now easily disable automated behaviours based on `getSettings().isPreviewMode` but doing this means that all the Nav blocks in the Patterns are empty. 

If this is somehow specific to Classic Menus as @scruffian alluded to in the description then I think we need to be clearer. For now I'm removing this from the priority list on the Tracking Issue as there doesn't appear to be much movement on it or requests for resolution.
